File: Grip/Package.m
//
//
// Package
// This is an object created by COBGIN
//
//
// by Mickey Barboi
//
//


#import "Package.h"

@implementation Package
@synthesize name, discount, order_index, group_id, created_at, updated_at;

#pragma mark Connection Manager
+ (RKObjectMapping *) mapping {
	//returns the mapping needed by RestKit to perform API calls
	RKObjectMapping* mapping = [RKObjectMapping mappingForClass:[self class]];
	[mapping addAttributeMappingsFromDictionary:@{
	@"name": @"name",
	@"discount": @"discount",
	@"order_index": @"order_index",
	@"group_id": @"group_id",
	@"created_at": @"created_at",
	@"updated_at": @"updated_at"  }];

	return mapping;
}


#pragma mark NSCoding Protocol
-(void)encodeWithCoder:(NSCoder *)encoder {
	[encoder encodeObject:name forKey:@"name"];
	[encoder encodeObject:discount forKey:@"discount"];
	[encoder encodeObject:order_index forKey:@"order_index"];
	[encoder encodeObject:group_id forKey:@"group_id"];
	[encoder encodeObject:created_at forKey:@"created_at"];
	[encoder encodeObject:updated_at forKey:@"updated_at"];
}

- (id)initWithCoder:(NSCoder *)decoder {
	self = [super init];
	self.name = [decoder decodeObjectForKey:@"name"];
	self.discount = [decoder decodeObjectForKey:@"discount"];
	self.order_index = [decoder decodeObjectForKey:@"order_index"];
	self.group_id = [decoder decodeObjectForKey:@"group_id"];
	self.created_at = [decoder decodeObjectForKey:@"created_at"];
	self.updated_at = [decoder decodeObjectForKey:@"updated_at"];
	return self;
}



@end

File: Grip/Package.swift
//
//  User.swift
//  Grip
//
//  Created by Mickey Barboi on 8/6/14.
//  Copyright (c) 2014 Mickey Barboi. All rights reserved.

import Foundation

class Package : MTLModel, MTLJSONSerializing {
    var created_at: NSDate?
    var discount = -1
    var group_id = -1
    var id = -1
    var name: String?
    var order_index: String?
    var updated_at: NSDate?

    var products = Array<Product>()
    
    //Boilerplate Mantle code
    class func appURLSchemeJSONTransformer() -> NSValueTransformer {
        return NSValueTransformer(forName: MTLURLValueTransformerName)!
    }
    
    class func JSONKeyPathsByPropertyKey() -> [NSObject : AnyObject]! {
        return [
            "products": NSNull()
        ]
    }
    
    class func JSONTransformerForKey(key: String) -> NSValueTransformer? {
        if key == "created_at" || key == "updated_at" {
            return NetworkUtils.dateFormatter()
        }
        
        return nil
    }

    
    //Boilerplate, compulsory overrides. Kinda stupid, isn't it?
    //The following four methods allow this class access to its superclass' inherited methods
    override func encodeWithCoder(coder: NSCoder!) {
        super.encodeWithCoder(coder)
    }
    
    required init(coder:NSCoder) {
        super.init(coder: coder)
    }

    override init() {
        super.init()
    }
    
    override init(dictionary dictionaryValue: [NSObject : AnyObject]!, error: NSErrorPointer) {
        super.init(dictionary: dictionaryValue!, error: error)
    }
}